### PR TITLE
GAE example

### PR DESCRIPTION
**Pull request by dmdashenkov**
- Added example of using GAE based storage (see `GaeApplication`)
 - Added tests for `GaeApplication` (a test is `@Ignore`d here; see [this issue](https://github.com/SpineEventEngine/gae-java/issues/5) for more details)
 - Also returned a _non blocking client_ example in `ClientApp`, removed in #3

### CODE REVIEW

**Comment by dmdashenkov** [comment]
@armiol PTAL

File: server-examples/build.gradle [hunk 1]
```diff
@@ -2,7 +2,7 @@
 dependencies {
     compile group: 'org.spine3', name: 'client', version: spineVersion
     compile group: 'org.spine3', name: 'server', version: spineVersion
-
+    compile group: 'org.spine3.gae-java', name: 'gcd', version: '0.2'
```
Comment (armiol) [inline server-examples/build.gradle]: Why `0.2`? We will need to reference the `0.6`.

Also, the group value is weird. The correct group is `org.spine3`, the name is `gae-java`. I am not sure about `gcd` — I just don't know what it is.

File: server-examples/src/main/java/org/spine3/examples/aggregate/ClientApp.java [hunk 2]
```diff
@@ -63,7 +69,44 @@
```
Comment (armiol) [inline server-examples/src/main/java/org/spine3/examples/aggregate/ClientApp.java]: Please make sure to append yourself as an `@author`.

Also, please review the methods of this class. Some may become `static`. For instance, `generateRequests` looks `static`.

File: server-examples/src/main/java/org/spine3/examples/storage/gae/GaeApplication.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        final LocalDatastoreStorageFactory storageFactory = LocalDatastoreStorageFactory.getDefaultInstance();
+        storageFactory.getOptions()
+                      .setEventIteratorPageSize(ITERATOR_PAGE_SIZE);
+        storageFactory.setUp();
+        final GaeApplication app = new GaeApplication(storageFactory);
+        app.execute();
+        storageFactory.tearDown();
+    }
+}
+
```
Comment (armiol) [inline server-examples/src/main/java/org/spine3/examples/storage/gae/GaeApplication.java]: Looks like an excessive empty line.

File: server-examples/src/main/java/org/spine3/examples/storage/gae/GaeApplication.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+     */
+    public GaeApplication(DatastoreStorageFactory storageFactory) {
+        super(storageFactory);
+    }
+
+    /**
+     * The entry point of the sample.
+     * <p>
+     * See instructions on configuring local Datastore environment on this page:
+     * https://github.com/SpineEventEngine/gae-java/wiki/Configuring-Local-Datastore-Environment
```
Comment (armiol) [inline server-examples/src/main/java/org/spine3/examples/storage/gae/GaeApplication.java]: Please rewrite into `The instruction on configuring the local Datastore environment is available <a href="[link address]">here.</a>`.

File: server-examples/src/main/java/org/spine3/examples/storage/gae/GaeApplication.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+import org.spine3.server.storage.datastore.LocalDatastoreStorageFactory;
+
+import java.io.IOException;
+
+/**
+ * A sample application showing basic usage of Spine framework running on GAE local Datastore.
+ *
+ * @author Alexander Litus
+ */
+public class GaeApplication extends Server {
```
Comment (armiol) [inline server-examples/src/main/java/org/spine3/examples/storage/gae/GaeApplication.java]: Maybe it's a `GaeServer`?

File: server-examples/src/main/java/org/spine3/examples/storage/gae/GaeApplication.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+import org.spine3.examples.aggregate.server.Server;
+import org.spine3.server.storage.datastore.DatastoreStorageFactory;
+import org.spine3.server.storage.datastore.LocalDatastoreStorageFactory;
+
+import java.io.IOException;
+
+/**
+ * A sample application showing basic usage of Spine framework running on GAE local Datastore.
+ *
+ * @author Alexander Litus
```
Comment (armiol) [inline server-examples/src/main/java/org/spine3/examples/storage/gae/GaeApplication.java]: Please make sure you are the `@author` as well.

File: server-examples/src/test/java/org/spine3/examples/storage/gae/GaeApplicationShould.java [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+                    server.awaitTermination();
+                } catch (IOException e) {
+                    throw propagate(e);
+                }
+            }
+        });
+
+        serverThread.start();
+
+        //noinspection ZeroLengthArrayAllocation
```
Comment (armiol) [inline server-examples/src/test/java/org/spine3/examples/storage/gae/GaeApplicationShould.java]: Either use a `@SuppressWarnings` or remove at all.

Comment-based suppressions do not affect the inspectors, only the IDEA hints.

**Comment by dmdashenkov** [comment]
@armiol I've addressed all your comments. PTAL again

**Comment by armiol** [comment]
@dmdashenkov LGTM.

**Pull request merged**